### DISCUSSION TITLE
All the wildcards disappeared from the Wildcards Manager tab despite the fact that the folders exist in Windows

### DISCUSSION THREAD

**Discussion by consecutiotemporum** [adieyal/sd-dynamic-prompts#331] [Q&A] [answered]
Today I launched the A1111 Web UI and found no wildcards in the "Wildcards Manager" tab, although I had five .txt files in the appropriate folder.

I removed the extension from the web UI, deleted the "\A1111 Web UI Autoinstaller\stable-diffusion-webui\extensions\sd-dynamic-prompts" folder in Windows, and I repeated the installation process.

There's the "Dynamic Prompts" in the txt2img tab, there's the "Wildcards Manager" tab, but although I have wildcards .txt in the Windows folder "\A1111 Web UI Autoinstaller\stable-diffusion-webui\extensions\sd-dynamic-prompts\wildcards", in the "Wildcards Manager" tab nothing appears anymore.

![Immagine 2023-03-25 103122](https://user-images.githubusercontent.com/114416380/227709728-85865c90-59e5-4d00-aec8-f4c1d5537749.jpg)

I tried copying one of the proposed Collections, for example "artists". The folders are correctly created, but they do not appear in the "Wildcards Manager" tab even if I restart the UI or A1111.

Thx.  

![Screenshot 2023-03-25 103019](https://user-images.githubusercontent.com/114416380/227709709-b64ce029-aec0-47ad-88a4-c7eb763ab739.jpg)

**Comment by adieyal** [comment]
Strange - any errors in your console?

**Reply by consecutiotemporum** [reply]
Thx, adieyal
I don't seem to see any.

![Immagine 2023-03-25 113210](https://user-images.githubusercontent.com/114416380/227711939-8e2dd996-8e83-44f0-994d-d9062074d0c5.jpg)

**Comment by adieyal** [comment]
Do the wildcards work in your prompts? i.e does `__rococco__` work for example?

**Reply by consecutiotemporum** [reply]
I think not.

prompt:
`__artists/genre/anime__`, red hair woman in white dress holding a glass of wine in her hand and not looking at the camera with a smile, looking left, in park, light from behind golden hours, ultrafine detailed

negative:
poorly drawn face, poorly drawn eyes, (out of focus), missing limbs, distortion, low resolution, missing hands, distorted hands, low detail, censored, deformed hands, crossed eyes, bad focal length, blurry, out of focus, out of frame, bad framing, tiling, long neck, extra heads, bad anatomy, bad proportions, extra limbs, extra torso, extra fingers, mutated hands, poorly drawn hands, 


I used a simple model SD15NewVAEpruned.ckpt [27a4ac756c] for the test

![00054-2886905005](https://user-images.githubusercontent.com/114416380/227725551-95b8000e-b764-42d1-8f46-3741f879d392.png)

**Comment by adieyal** [comment, marked as answer]
I'm not sure why that's happening. I can't re-create it on my side. Are you using the latest version? It probably won't help, but worth making sure. 

I don't think it's a bug, rather a configuration problem. Double checked your path, in case there is a minor difference.
In lieu of that, we will soon release a re-design of the wildcard manager which might address any lingering bugs.

Sorry I can't be of more help.

**Reply by consecutiotemporum** [reply]
You have been exceedingly kind. I will try to recheck, although everything seems fine.
For the time being, I'll mark the thread as answered so you don't get confused.
Thx.

**Reply by gaxyope** [reply]
hi adieyal.
i have the same problem as OP
this is my error in the console
Launching Web UI with arguments: --autolaunch --medvram --xformers
Error loading script: dynamic_prompting.py
Traceback (most recent call last):
  File "F:\SD\SD 1.5\stable-diffusion-webui\modules\scripts.py", line 256, in load_scripts
    script_module = script_loading.load_module(scriptfile.path)
  File "F:\SD\SD 1.5\stable-diffusion-webui\modules\script_loading.py", line 11, in load_module
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "F:\SD\SD 1.5\stable-diffusion-webui\scripts\dynamic_prompting.py", line 8, in <module>
    import dynamicprompts
ModuleNotFoundError: No module named 'dynamicprompts'

kinda new to to this stuff so i dont know what to do at the moment on how to fix this,
appreciate the help.

**Reply by adieyal** [reply]
for some reason your dynamicprompts library is not installed. Try pip install dynamicprompts==0.10.5 
are you using the latest version of the extension?

**Reply by gaxyope** [reply]
959006f4 (Sat Mar 25 14:45:23 2023) | latest

Yes it is on the latest version.
However I did re-read your reply to OP and the wildcards DOES work in my prompt although it is totally gone from the web ui.
I will use that method for now.
Thanks for the help.

**Reply by hotkeysoft** [reply]
It doesn't work for me either. I deleted the whole extension and reinstalled it 
version: 2c1af624 (Tue Mar 28 09:40:52 2023) 

Wildcard Manager tab is still empty. No errors in console.

**Comment by adieyal** [comment]
Have a look at #338 - I have posted a patch which I think fixes the issue.

**Reply by chewtoys** [reply]
its still empty with latest update 

![image](https://user-images.githubusercontent.com/9331843/228689545-041fb2bd-9893-4dc8-96e9-911d9d3f3323.png)

**Reply by adieyal** [reply]
Could you confirm your dynamicprompts version? You can find it on the Need Help accordion

**Reply by chewtoys** [reply]
![image](https://user-images.githubusercontent.com/9331843/228722041-f69ec913-1e91-4eef-ad5b-d2aa21165a67.png)

**Comment by BlastedRemnants** [comment]
Hello, I'm just dropping by to say that my wildcards also don't show up in the Manager. I used to always have them in a subfolder named "mine", and it always worked like that. Now tho, they don't show in Manager at all and don't work either (console shows "no values found for wildcard x") unless I pull them up out of the subfolder and drop them all loose into the "wildcards" folder. Then I can call them in my prompt, and the auto-complete can read them as well, but they still don't show up in the manager. According to the info given to me in the accordion thing I've got "version 2.10.3 of the WebUI extension, and the underlying dynamicprompts library is version 0.21.2"

**Reply by chewtoys** [reply]
yep, the issue is still not fixed, i reported it, and even found the cause of this issue.

https://github.com/adieyal/sd-dynamic-prompts/issues/359#issuecomment-1489724148

and 

https://github.com/adieyal/sd-dynamic-prompts/issues/360

**Reply by BlastedRemnants** [reply]
Thank you good sir and/or madam! For me it was https://github.com/bbc-mc/sdweb-merge-block-weighted-gui that was causing the issue, I disabled it and everything is working as expected now. Many thanks, wherever you're from you're a national treasure hahaha  :D

**Reply by hahajaja** [reply]
Conflicts with this plugin https://github.com/hako-mikan/sd-webui-supermerger

**Reply by adieyal** [reply]
have you tried v2.10.4? That tightened up the javascript to prevent incompatibilities between extensions. Let us know if you're still enountering these issues with that version.

**Reply by BlastedRemnants** [reply]
> have you tried v2.10.4?

Sorry for the late reply, I saw your question but thought maybe it was meant for someone else. In any case yes I did, I check my extensions everytime I launch and I've also gotten rid of any that cause issues with this one. I still have a problem tho, the Manager shows my folder and lists the wildcards inside it, and I can edit them from within the UI and everything seems to be working. When I generate a picture tho the console shows "No values found for wildcard x" with x being the name of my wildcard file, and the pictures I get do not use the prompts from my wildcards. The only way I can make my wildcards work is to move them out of the subfolder and keep them in the wildcards folder, which gets pretty messy in a hurry but otherwise works just fine.

**Reply by bencoster** [reply]
Hi @adieyal, I can confirm as of today 12/04/2023 what @hahajaja has found and that is a plugin https://github.com/hako-mikan/sd-webui-supermerger) version 7aaa6a94 (Tue Apr 11 13:17:07 2023) prevents Wildcards Manager from populating wildcards found in the the correct location, basically kills the Wildcard manager. Although the Wildcards used already in prompts or if you manually add them work. The sd-dynamic-prompts - Wildcard manager version I am using is 707fa9a1 (Thu Apr 6 18:31:57 2023) and it will populate/list the Wildcard folders under "4. Optional - add your own wildcards files to the C:\Vid-Extract\stable-diffusion-webui\extensions\sd-dynamic-prompts\wildcards folder" of the Wildcards Manager, only if I deselect the tick box for the sd-webui-supermerger in the Automatic1111 Extensions Installed tab. 

I am running Automatic1111 with these versions:
python: 3.10.8  
torch: 1.13.1+cu117  
xformers: 0.0.16rc425  
gradio: 3.23.0  
commit: 22bcc7be  
checkpoint: 980c17bc33
OS: Windows 11

Thanks for the great tool adieyal and providing updates.

**Reply by TarryZu** [reply]
@adieyal Exactly, when I removed sd-webui-supermerger according to @hahajaja 's practice, the directory tree in wildcards manager appeared magically. Of course, I don't know what caused the conflict, json?

**Comment by tenabraex** [comment]
![Uploading wildcard2.png…]()
Ive just had this issue pop up this week after the most recent update. WIldcards work fine, they just don't display in the wildcards manager
![wildcard](https://user-images.githubusercontent.com/47406324/233763965-30f9814c-3508-4ddb-9f1e-0a444831dee4.png)

**Comment by adieyal** [comment]
It may be due to a conflict with another  extension. Try the following:

1. Have a look in the developer console of your browser to see if there are any javascript errors. 
2. Try turning off other extensions to see if that helps. If you find a conflict let me know.

let me know if you find anything

**Reply by poisenbery** [reply]
Well this sucked!
I had to disable every extension, and re-enable 1 by 1

I have found that the Merge Block Weight extension is causing this conflict.

https://github.com/bbc-mc/sdweb-merge-block-weighted-gui

**Reply by chewtoys** [reply]
It never was fixed, I reported it long ago, https://github.com/adieyal/sd-dynamic-prompts/issues/359#issuecomment-1489724148

And it was moved here then forgotten about it https://github.com/adieyal/sd-dynamic-prompts/issues/360

**Reply by poisenbery** [reply]
> It never was fixed, I reported it long ago, [#359 (comment)](https://github.com/adieyal/sd-dynamic-prompts/issues/359#issuecomment-1489724148)
> 
> And it was moved here then forgotten about it #360

The solution I read was "figure it out yourself" lmao
Was "fun" >_>

Why was it marked "Answered" when it's clearly not been fixed???
